metricbeat/helper/prometheus/textparse.go

```diff
@@ -496,8 +498,15 @@ func ParseMetricFamilies(b []byte, contentType string, ts time.Time) ([]*MetricF
 		)
 		if et, err = parser.Next(); err != nil {
 			// TODO: log here
-			// if errors.Is(err, io.EOF) {}
-			break
+			if errors.Is(err, io.EOF) {
```

Comment:
I add here the comment from Tania. I agree is important because now we dont know 
Import 	"github.com/elastic/elastic-agent-libs/logp"

Possible suggestion:
```
if et, err = parser.Next(); err != nil {
			if errors.Is(err, io.EOF) {
				log.Errorf("EOF is reached: %v ", err)
				break
			}
			log.Debugf("Parse Prometheus Family error : %v ", err)	
			break
		}
```

Reply:
thanks for the suggestion. 

I am a bit skeptic about using a global logger and I thought we had to pass prometheus logger to the ParseMetricFamilies instead.

I tested both solutions. They both work, I think it is about styling. The current changes use Kubernetes logger since that's the only other logger globally available.

if we wanted to use Prometheus logger there a few more changes required in both the code and the tests.

Reply:
Sorry got confused with what is global and prometheus logger. 

Currently I see that  `ParseMetricFamilies` is calle from GetStateMetricsFamilies which is used from state metricsets here :https://github.com/elastic/beats/blob/main/metricbeat/helper/kubernetes/state_metricset.go#L115

So if we pass the error back to the state_metricset.go in abolve logger this would be able to log it. Is not it? Just to make sure that we return it in ParseMetricFamilies

Do you agree Giuseppe? Then in this case we dont want to add a new logger

Reply:
Thank you @gsantoro ! Now I think I got it. We need to return "invalid metric type" message back.

Because we already have prometheus logger, I guess is preferrable to pass it from caller function and create another one

Reply:
Thank you for adding logging!
Can you please also add a test for this use case? That `info` is skipped but rest of metrics are parsed correctly for `ContentTypeTextFormat`